### ISSUE TITLE
FK constraint bug `(db_id)=(1)` on load-from-h2

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#33666]
**Is your feature request related to a problem? Please describe.**
Trying to load `1.46.6.4` snapshot into an instance running `1.46.8` running instance and it fails.
Likely related to the H2 security incident.

```
ERROR: insert or update on table \"metabase_table\" violates foreign key constraint \"fk_table_ref_database_id\"\n  Detail: Key (db_id)=(1) is not present in table \"metabase_database\".\n\tat 
```

**Describe the solution you'd like**
`load-from-h2` should work

**How to reproduce**
1. Start a new `46.6.4` instance: https://downloads.metabase.com/enterprise/v1.46.6.4/metabase.jar
2. Run `java -jar metabase.jar dump-to-h2 dumph2 --dump-plaintext` to get a dump file
3. On a different directory, start a new `46.8` instance with a **Postgres** database https://downloads.metabase.com/enterprise/v1.46.8/metabase.jar
4. Run `java -jar metabase.jar load-from-h2 path/to/dumph2`
5. See error

**How important is this feature to you?**
This is getting critical at this point: we can't automatically migrate to Cloud any instance on or below 46.6.4

**Comment by luizarakaki** [comment]
@Tony-metabase why can't we upgrade the source instance before running `load-from-h2`?

I don't think we should make this script compatible across multiple versions while the workaround is just upgrading the source instance

**Comment by luizarakaki** [comment]
In the end we would do the same thing, run upgrades or downgrades behind the scene. It is better to run these processes explicitly and handle problems accordingly.

**Comment by Tony-metabase** [comment]
This is getting critical for the two points mentioned above: 

1) Friction for customers wanting to migrate to cloud. We say it's seamless to migrate to cloud yet customers and lately there has been a lot end up opening tickets to ask for help on upgrades

2) If you take cloud for example when a backup is taken form an instance and then the instance gets upgraded it's practically useless. 

If the destination instance is already on the higher version and the source is simply the h2 snapshot, what is the issue with making the data compatible with the upgraded version? Since the schema information is already present in the destination version

**Comment by luizarakaki** [comment]
Excluding the security fix issue, you should be able to `load-from-h2` across minor versions.

**Comment by Tony-metabase** [comment]
Fixing that would already be helpful! Thank you :)

I do understand though that migrating across versions is not always possible since i don't expect the load-form-h2 to work when you are loading from an 47 dump into a 46 instance but why would the opposite be a problem? Since the Instance getting the dump already has all the schema information that needs to be applied to the dump. Since what is happening is an upgrade not a downgrade

**Comment by luizarakaki** [comment]
It is possible. We would need to run the upgrade process, but this makes things more complicated (we could do the same for downgrades too, we support downgrading now).

But we would add more complexity to an already complicated process. Maintaining upgrades and downgrades separated from migrations keep everything more isolated and explicit

**Comment by camsaul** [comment]
I wrote down the cause of this and how to fix it here https://metaboat.slack.com/archives/C04DN5VRQM6/p1696462307569299?thread_ts=1696451657.303989&cid=C04DN5VRQM6

**Comment by camsaul** [comment]
Duplicate of #32837, this was fixed by #32960 (47) and #32961 (46.7)
